src/source/extract.rs

```
@@ -136,9 +136,38 @@ pub(crate) fn extract_tar(
     ));
 
     let tmp_extraction_dir = tempfile::Builder::new().tempdir_in(target_directory)?;
-    archive
-        .unpack(&tmp_extraction_dir)
-        .map_err(|e| SourceError::TarExtractionError(e.to_string()))?;
+    for entry in archive.entries()? {
+        match entry {
+            Ok(mut file) => {
+                let path = tmp_extraction_dir.path().join(file.path()?);
+
+                // Ensure all intermediate directories exist before unpacking
+                if let Some(parent) = path.parent() {
+                    fs::create_dir_all(parent)?;
+                }
+
+                // Attempt to unpack symlinks and regular files, logging any issues
+                match file.header().entry_type() {
+                    EntryType::Symlink if cfg!(target_os = "windows") => {
+                        if let Err(e) = file.unpack(&path) {
+                            println!(
+                                "Warning: failed to extract symlink {:?} due to {:?}",
+                                path, e
+                            );
+                        }
+                    }
+                    _ => {
+                        if let Err(e) = file.unpack(&path) {
+                            println!("Warning: failed to unpack {:?} due to {:?}", path, e);
```

Again we should return the error here :)